py/dml/structure.py

```diff
@@ -2711,11 +2766,12 @@ def mkexpr(self, indices):
         return self.cached
 
 def mkparam(obj, autoparams, param):
-    _, site, name, (value, default, auto) = param
+    (_, site, name, type_info, is_default, value) = param
+    auto = type_info is not None and type_info.kind == 'auto'
 
     # auto-parameters are like normal "hard" assignments
     if name in autoparams:
-        if not auto and not default:
+        if not auto and not (dml.globals.dml_version == (1, 2) and is_default):
```

Comment:
If I'm interpreting this right, 1.2 abuses `default` declarations and possibly overrides them with autoparams? Wow.

Reply:
It's from `parameter indexvar default undefined;`.

BTW: This means that 1.2 incidentally allows user-provided definitions of autoparams, e.g.:
```
group g {
    parameter indexvar default 18;
    parameter bank default 3;
}
```
(where the indexvar decl will give `WEXPERIMENTAL: parameter with two default declarations` and have effect, whereas the bank decl will be silently swallowed and have no effect).

IIRC I once suggested to make it a proper auto but others (probably Jonatan) deemed that unimportant. And unimportant it is, but since you bothered to comment it I'll push another commit that cleans it up.